### PR TITLE
Add instructional 'Fonts' section to customizer when fonts are only modifiable from 'Global Styles'

### LINKED ISSUE #53473: Global Styles: Make it easier to find
Users running [most](https://github.com/Automattic/themes/issues/3202) themes on WordPress.com that **don't** have a Fonts panel in the Customizer instead have the option to change their Base & Heading font via a Global Styles panel in the post and page editor.

Edit_Post_‹_Global_Styles_Demo_—_WordPress_com

Unfortunately, many users entirely miss the fact that this icon is the key to finding the spot where they can modify their fonts site-wide, and end up either not changing their fonts and remaining frustrated, contacting support for help, or sometimes even churning.

I think the reasons they have trouble finding it are many fold, including:

- The editor is where people expect to **edit page and post content**; it's not intuitive to have site-wide design settings tucked there as well.
- The A icon is tiny and not immediately obvious as a typography/font control.
- The icon is alongside many other competing icons/links/buttons and easily missed.
- The "tooltip" wording Global Styles is not standard and not easily recognized by users. 
 
### Interim Suggestions

- Could we rename the "Global Styles" tooltip to "Fonts" for the time being – at least until there are other settings tucked in with it as Global Styles is developed further? 
- Could we add a Fonts panel in the Customizer for these themes, with a prompt pointing people to Global Styles? That way, when people go to the spot where they expect to find font options, there's some guidance as to where they can find the setting they're looking for. Very rough mockup:

Global Styles - Customizer prompt

I'm sure there are other ideas as well.

**Additional context**
Thousands of interactions with WordPress.com users in live chat, email, and Quick Start sessions.

### PR DESCRIPTION
**Pull request by DanielSmithMichigan**
---

[What users will see if we can't pick out a page to open the block editor with](https://user-images.githubusercontent.com/10274366/124637710-1540f900-de58-11eb-85a5-0fb0694ed784.png)

#### Simple Site Testing Instructions

1. Host a calypso app locally: `cd wp-calypso; yarn && yarn start`
2. Host the editing toolkit plugin as well. `cd wp-calypso/apps/editing-toolkit; yarn dev --sync`
3. Make sure your hosts file is pointing wordpress.com and public-api.wordpress.com to your sandbox, as well as the specific site you're about to test with.
4. Make sure you have an active SSH connection to your sandbox
4. Perform all the following steps using your locally hosted calypso app.
3. Activate a theme which requires fonts to be modified from global styles (such as twenty twenty one)
4. Go to customizer
5. Expand the 'Fonts' section
6. Read the content.
7. Click the link which says 'Click here to open the Block Editor and change your fonts.'
8. It should open your homepage and expand the global styles section
9. This tracks event should have been triggered: calypso_customizer_global_styles_block_editor_link_clicked
10. Click the link which says 'Learn more about changing fonts using Global Styles.'
11. It should open this page: https://wordpress.com/support/custom-fonts/#changing-fonts-with-global-styles
12. This tracks event should have been triggered: calypso_customizer_global_styles_support_link_clicked

#### Atomic Testing Instructions

1. Create an atomic site
2. Make sure your hosts file is pointing to production servers
3. Download the zipped version of the editing toolkit plugin from [here](https://teamcity.a8c.com/viewLog.html?buildId=6417514&buildTypeId=calypso_WPComPlugins_EditorToolKit&tab=artifacts#).
4. Install/activate this plugin (You may need to deactivate the existing editing toolkit plugin first)
3. Activate a theme which requires fonts to be modified from global styles (such as twenty twenty one)
4. Go to customizer
5. Expand the 'Fonts' section
6. Read the content.
7. Click the link which says 'Click here to open the Block Editor and change your fonts.'
8. It should open your homepage in the block editor (The global styles section won't auto expand because we're using the local development environment. But you should have ?openSidebar=globalStyles in the URL you were brought to)
9. This tracks event should have been triggered: calypso_customizer_global_styles_block_editor_link_clicked
10. Click the link which says 'Learn more about changing fonts using Global Styles.'
11. It should open this page: https://wordpress.com/support/custom-fonts/#changing-fonts-with-global-styles
12. This tracks event should have been triggered: calypso_customizer_global_styles_support_link_clicked

#### Test that the new 'fonts' section doesn't show if there is an existing 'fonts' section

1. Host a calypso app locally: `cd wp-calypso; yarn && yarn start`
2. Host the editing toolkit plugin as well. `cd wp-calypso/apps/editing-toolkit; yarn dev --sync`
(NOTE: If you run into any trouble getting toolkit setup, you can refer to [this readme](https://github.com/Automattic/wp-calypso/tree/trunk/apps/editing-toolkit) for an additional information)
3. Make sure your hosts file is pointing wordpress.com to your sandbox, as well as the specific site you're about to test with.
4. Make sure you have an active SSH connection to your sandbox
4. Perform all the following steps using your locally hosted calypso app.
1. Activate a theme which does not support global styles (such as twenty twenty)
2. Go to customizer
3. The 'Fonts' section described above should not exist

https://user-images.githubusercontent.com/10274366/124625215-9cd43b00-de4b-11eb-9c2d-5a79da511bec.mov

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php]: What are the benefits of defining a class in a function?

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php]: Could you add some sanitization here? `esc_url_raw()` is probably our best bet.
https://developer.wordpress.org/reference/functions/esc_url_raw/
https://codex.wordpress.org/Validating_Sanitizing_and_Escaping_User_Data#Sanitizing:_Cleaning_User_Input

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php]: There's an escaping function that can help you here, `esc_url()`.
See https://developer.wordpress.org/reference/functions/esc_url/

**Review by obenland — COMMENTED** [review]
@DanielSmithMichigan Do you have `phpcs` set up locally? It helps tremendously with catching non-obvious details: PCYsg-sax-p2

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php
Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php]: Basically I was adding conditional logic to prevent the class definition from being created more than once, if the file was included more than once.

I don't have a reason we can't just use `if ( ! class_exists )` without defining a function - I can switch over to that before this is complete.

**Comment by DanielSmithMichigan** [comment]
> @DanielSmithMichigan Do you have `phpcs` set up locally? It helps tremendously with catching non-obvious details: PCYsg-sax-p2

Linting runs on commit for me (which I had explicitly ignored for now). I'll checkout the vscode plugin, hopefully that will help me deal with it prior to commit.

I wasn't too confident in the high level details of this implementation, so yesterday my goal was just to get something up here that could be reviewed, and with testing instructions as soon as possible.

But now that there's some structure here, I'm can work on the details today.

**Comment by creativecoder** [comment]
@DanielSmithMichigan In case it helps... this VS Code extension has been working the best for me, recently: https://github.com/valeryan/vscode-phpsab

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Since it's you who added the `<p>` tags, the `wp_kses` call is probably not necessary.

If you want to make sure that translators can't alter the markup structure, you could use [`esc_html__()`](https://developer.wordpress.org/reference/functions/esc_html__/)

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Props for the translator comment!

We generally try to avoid having them translate markup as much as possible (there are always exceptions of course). In this case you could have the anchor be in a regular string, introduce a second placeholder and pass the translation function with the anchor text as a third parameter to `sprintf()`.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Same thing as above, if you separated out the translation string from the anchor you don't need to pass it through `wp_kses()`

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (obenland) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php:107]: Unless the link to the support page is something translators should translate to point to localized versions, it probably doesn't need to be translated.

If it does, I'd probably still separate it out and add a translator comment to make it clear where it points to and what the purpose is, just to keep those translation strings as simple as possible

**Review by obenland — COMMENTED** [review]
Nice work @DanielSmithMichigan!

I added a few comments about simplifying translation strings and their escaping, I hope it helps!

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/src/global-styles-sidebar.js
Comment (kwight) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/src/global-styles-sidebar.js]: Yes, we can record Tracks events server-side too: PCYsg-4KT-p2

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/src/global-styles-sidebar.js]: Thanks. Tracks events are being reported from javascript functions now, and those functions are constructed serverside. I like the implementation much more now.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: To be honest, I didn't think the output sanitization was needed for the same reason, but I was receiving linter warnings due to not escaping HTML before output. It seems like maybe that was just due to the fact that I had stored the HTML in a variable. I've moved stuff around a little bit based on your suggestions, and I used esc_html on just the translated content, and I'm happier with where everything has ended up now.

**Comment by kwight** [comment]
This is looking good @DanielSmithMichigan , but the Global Styles panel doesn't open for me:



I'm following the instructions at PCYsg-ly5-p2 to test (using the `install-plugin.sh` command), and have the Barnsbury theme activated on a sandboxed test site.

**Comment by DanielSmithMichigan** [comment]
> This is looking good @DanielSmithMichigan , but the Global Styles panel doesn't open for me:
> 
> I'm following the instructions at PCYsg-ly5-p2 to test (using the `install-plugin.sh` command), and have the Barnsbury theme activated on a sandboxed test site.

I think the issue is that the change here: client/gutenberg/editor/calypsoify-iframe.tsx is a calypso change, and you're loading from wordpress.com, so that change isn't loaded for you. The "openSidebar" parameter is not passed into the iframe in which it's needed.

Feel free to pull me into a call. I'll update the testing steps for how I was loading the app as well.

**Comment by kwight** [comment]
> I think the issue is that the change here: client/gutenberg/editor/calypsoify-iframe.tsx is a calypso change

Doh! Dangit, you're right, that was it – works great now. 🎉 

I noticed that this Global Styles plugin is marked as deprecated (to be replaced by core), so I'm even questioning myself if Editing Toolkit is the most appropriate place to put this code (another option is committing to a dotcom sandbox for Simple and `wpcomsh` for Atomic). @obenland do you have and preferences/concerns either way?

**Comment by mreishus** [comment]
Everything is working for me, including opening the global styles sidebar.


Deferring to others for the conversation about where it should be placed.

**Comment by kwight** [comment]
> so I'm even questioning myself if Editing Toolkit is the most appropriate place to put this code

@DanielSmithMichigan let's ignore this comment for now; your code is working well, Konstantin is AFK today, and if we really need to change it, it can be done as a later followup. Let's get this in once we can confirm it's behaving as expected on Atomic.

**Comment by mreishus** [comment]
I have not tested on atomic yet. I suppose it's the same directions instead I need to sync the editing toolkit changes to an atomic site, trying that..

**Comment by mreishus** [comment]
I am not having much luck on atomic, but I very well could be testing wrong. 

Locally, in `wp-calypso/apps/editing-toolkit`, I ran `yarn build`. Which generated some files.. I think in `wp-calypso/apps/editing-toolkit/editing-toolkit-plugin/global-styles/dist`.

I set up an atomic dev site. I sshed to it and looked for editing-toolkit, but I could only find `~/wp-content/plugins/full-site-editing` (the old name). Well, there's a `~/wp-content/plugins/full-site-editing/global-styles/dist` directory, so I moved the existing files out of it and scped over the ones I made in the previous step.

Now the customizer fails to load. Checking the error log I see `PHP Fatal error:  Uncaught Error: Class 'WPCOM_Masterbar' not found in /srv/htdocs/wp-content/plugins/full-site-editing/global-styles/class-global-styles-fonts-message-control.php:29`. 

I do not know if this reflects an actual error that would happen on atomic, an error in my testing procedure, or something else.

Edit: Actually, I copied over some files outside of the `dist` directory as well, as my error message reveals.

**Comment by mreishus** [comment]
The latest set of changes for me is now working on atomic and continues to work on simple :+1:


Holding off on the official :heavy_check_mark: since @DanielSmithMichigan noticed the tracks events weren't working.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (creativecoder) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Don't forget that we have horizon.wordpress.com, and wpcalypso.wordpress.com, as well.

I think you can use `WPCOM_Masterbar::get_calypso_site_slug`, but it would need an `exists` check with a fallback, since it's only available on wpcom. For example: https://github.com/Automattic/jetpack/blob/529d07f441bb4667acd773918563296878e19cb4/projects/packages/status/src/class-status.php#L268

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: @creativecoder @kwight With regard to horizon.wordpress.com and wpcalypso.wordpress.com,

Is this something that we can follow up with right after deployment?

If we leave it out for right now, it just means if someone clicks this block editor link, they will be directed to the wordpress.com version.

Comment (kwight) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: @DanielSmithMichigan Let's do the `WPCOM_Masterbar::get_calypso_site_slug` fix now, as well as verifying Tracks are firing before merging. It's very close!

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: @kwight Yep I just pushed the WPCOM_Masterbar::get_calypso_site_slug fix.

The tracks events are firing as well.

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Just waiting for the atomic artifact to finish building. And I'll try atomic one more time to double check against my most recent changes. Then we should be good to go.

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Everything is working on atomic. I don't have anything on my plate for this PR at this moment.

**Review by mreishus — DISMISSED** [review]
Retested on simple and atomic and everything is looking good :+1:. I'm ok with waiting until the next iteration to ensure that we're not rushing the ETK deploy 'under the wire'.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (kwight) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php:87]: Should this all be handled with `get_site_slug` now?

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (kwight) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: I admit using concatenation for piecing together a script worries me, as it can be very fragile. It would be better to enqueue a standalone script, which is more "the WordPress way", and expected by other WP devs. https://developer.wordpress.org/reference/functions/wp_enqueue_script/

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (kwight) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: I don't think any of this is necessary, isn't it handled automatically by the script when the user is logged in? PCYsg-4KT-p2

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (kwight) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php:87]: Ah, I see. `calypsoOrigin` will only exist in the iframed Calypso context. In wp-admin, you'd need to get the host with `get_blog_details()->domain`.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/static/attach-tracks-events-to-fonts-links.js
Comment (DanielSmithMichigan) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/static/attach-tracks-events-to-fonts-links.js]: Thank you for taking a second look at this pull request, @kwight.

The reason my earlier attempts at this javascript file didn't work: I didn't originally realize that I would need to interact with the parent iframe to find the DOM element.

But one concern I've left with: This file isn't babel transpiled right now. I'm concerned since most javascript files are being served through the dist folder, and I'm breaking that standard.

But in order to babel transpile this file, I will basically need to create a whole new package to be built. I already got that working before, so I think I could do it quickly. I'm just concerned that maybe it's not a good idea since it might introduce a lot of unexpected complexity somewhere along the line (maybe even after merging)? Or maybe it's not a good idea since this will be deprecated? Just wanted to get your thoughts.

The package definitions are in this [package.json](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/package.json). We use an npm-run-all script to run through every package in that file. Lines 28 -> 31 are an example of how we achieve multiple packages in one folder.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: Moving to an enqueued script resolved this, thank you!

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php
Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php]: I agree, it's a mess. I was rushing when my previous implementation didn't test well on atomic right before the sprint demo. The newer version should be better, I hope.

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/static/attach-tracks-events-to-fonts-links.js
Comment (kwight) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/static/attach-tracks-events-to-fonts-links.js]: We discussed this, and decided that doing it properly was the way to go, even if it means a new bundle.

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/static/attach-tracks-events-to-fonts-links.js]: This should be all set

Comment (DanielSmithMichigan) [reply apps/editing-toolkit/editing-toolkit-plugin/global-styles/static/attach-tracks-events-to-fonts-links.js]: Actually hold off one minute...

**Review by kwight — APPROVED** [review]
:shipit: Works great on Atomic and Simple, no problems at all! 🎉

**Pull request merged**
